### PR TITLE
Add health check for WaggleDance service

### PR DESCRIPTION
**Pull request by pradeepbhadani**

### CODE REVIEW

File: templates/waggledance.json [hunk 1]
```diff
@@ -34,6 +34,12 @@
        "name": "BASTION_SSH_KEY_ARN",
        "value": "${bastion_ssh_key_arn}"
      }
-    ]
+   ],
+   "healthCheck": {
+       "command": ["CMD-SHELL", "curl -f http://localhost:18000/actuator/health || exit 1"],
+       "interval": 5,
```
Comment (massdosage) [inline templates/waggledance.json]: Just checking that this is measured in seconds and not milliseconds ;)

Comment (rpoluri) [reply templates/waggledance.json]: What happens when that URL is working but returning failure code?
I assume it only exist with code 1 when service is not responding on that port.

Ideally we should configure route 53 service discovery health check and check for http return codes.
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-discovery.html

Comment (mroark-exp) [reply templates/waggledance.json]: The "-f" flag handles that part, from: https://curl.haxx.se/docs/manpage.html

> -f, --fail

> (HTTP) Fail silently (no output at all) on server errors. This is mostly done to better enable scripts etc to better deal with failed attempts. In normal cases when an HTTP server fails to deliver a document, it returns an HTML document stating so (which often also describes why and more). This flag will prevent curl from outputting that and return error 22.

> This method is not fail-safe and there are occasions where non-successful response codes will slip through, especially when authentication is involved (response codes 401 and 407).

**Comment by mroark-exp** [comment]
@rpoluri not sure if this is relevant here: https://github.com/aws/amazon-ecs-agent/issues/1421

**Comment by mroark-exp** [comment]
The only thing I can find in the route 53 docs about what happens if route 53 health check fails is that it triggers cloudwatch and DNS failover. But that won't cause the underlying ECS task to get replaced right? The ECS healthcheck failing should cause the ECS task to get replaced, right?

So it seems to me maybe we need both actually? But based on the bug report above, it almost sounds like route 53 is ignoring ECS health anyway.

>"ECS Service Discovery registering is ignoring Taskhealth check and adding A records in pool even task health check is failing"

**Comment by mroark-exp** [comment]
I'm also confused about this verbiage:

https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-discovery.html

> When specifying health checks for your service discovery service, you must use either custom health checks managed by Amazon ECS or Route 53 health checks. The two options for health checks cannot be combined. 

We specified health_check_custom_config in our metastore_proxy resource, and it says, "HealthCheckCustomConfig—Amazon ECS manages health checks on your behalf." So I guess route 53 checks won't even work?

**Pull request merged**